File: src/common/search_result_item.rs
use std::borrow::Cow;

use super::icon::Icon;
use fuzzy_matcher::skim::SkimMatcherV2;
use serde::Serialize;

#[derive(Serialize, Debug, Clone)]
pub struct SearchResultItem<'a> {
    pub primary_text: Cow<'a, str>,
    pub secondary_text: Cow<'a, str>,
    pub icon: Icon<'a>,
    pub needs_confirmation: bool,
    pub id: Cow<'a, str>,
    pub score: i64,
}

pub trait IntoSearchResultItem {
    fn fuzzy_match(&self, query: &str, matcher: &SkimMatcherV2) -> Option<SearchResultItem>;
}

File: kal-ui/src/search_result_item.ts
export interface Icon {
  data: string;
  type: IconType;
}

export enum IconType {
  Path = "Path",
  Svg = "Svg",
  BuiltIn = "BuiltIn",
  Url = "Url",
}

export interface SearchResultItem {
  primary_text: string;
  secondary_text: string;
  icon: Icon;
  needs_confirmation: boolean;
  id: string;
}
